Sources/MassTransit/MassTransitUtils.swift

```diff
@@ -0,0 +1,22 @@
+import AsyncAlgorithms
+
+func randomString(length: Int) -> String {
+    let letters = "abcdefghijklmnopqrstuvwxyz0123456789"
+    var randomString = ""
+    for _ in 0..<length {
+        let randomIndex = Int.random(in: 0..<letters.count)
+        let letter = letters[letters.index(letters.startIndex, offsetBy: randomIndex)]
+        randomString += String(letter)
+    }
+    return randomString
+}
+
+func getModuleName<T>(_ module: T) -> String {
+    return String(String(reflecting: T.self).prefix { $0 != "." }).replacingOccurrences(of: "_", with: "")
+}
+
+func gracefulCancellableDelay(timeout: Duration) async throws {
+    for await _ in AsyncTimerSequence(interval: timeout, clock: .continuous).cancelOnGracefulShutdown() {
+        break
+    }
+}
```

Comment:
The function is declared to be throwing, but it doesn't. I wonder if it should, though. I would expect the same semantics as Task.sleep: throwing on interruption.

Reply:
Well, the AsyncTimerSequence doesn't throw, so I guess the top method doesn't need to throw either!

Reply:
I have not really looked at where this is used, but in general the problem I see with that is that the caller cannot know whether the specified time has passed or the function returned early.